drivers/timer/stm32_lptim_timer.c

```diff
@@ -67,6 +67,9 @@ static int32_t lptim_time_base;
 /* The prescaler given by the DTS and to apply to the lptim1 clock */
 #define LPTIM_CLOCK_RATIO DT_PROP(DT_NODELABEL(lptim1), st_prescaler)
 
+/* Actual lptim clock freq when the clock source is reduced by the prescaler */
+static uint32_t lptim_clock = 32000;
```

Comment:
Use of `lptim_clock` name vs `lptim_clock_freq` is confusing.
This does not explain the difference between the 2 values, and why sometime `lptim_clock_freq / LPTIM_CLOCK_RATIO` should be used instead of `lptim_clock_freq`.

What about `lptim_set_autoreload((lptim_clock_freq / CONFIG_SYS_CLOCK_TICKS_PER_SEC) - 1);`